pkg/bmcpfs/controllerserver.go

```diff
@@ -61,18 +113,153 @@ func newControllerServer(region string) (*controllerServer, error) {
 	if err != nil {
 		return nil, err
 	}
+	kubeconfig := mustGetKubeClient()
+	nodeInformer := informerv1.NewFilteredNodeInformer(kubeconfig, 0, cache.Indexers{NODEID_INDEX: nodeIndexFunc}, func(options *metav1.ListOptions) {
+		options.FieldSelector = fields.OneTermEqualSelector("alibabacloud.com/lingjun-worker", "true").String()
+	})
+	nodeInformer.SetTransform(nodeTransformFunc)
 	return &controllerServer{
 		vscManager:     internal.NewPrimaryVscManagerWithCache(efloClient),
 		attachDetacher: internal.NewCPFSAttachDetacher(nasClient),
+		filsetManager:  internal.NewCPFSFileSetManager(nasClient),
 		nasClient:      nasClient,
+		nodeInformer:   nodeInformer,
 		skipDetach:     skipDetach,
+		bmcpfsLocks:    utils.NewVolumeLocks(), // 重新初始化锁表
 	}, nil
 }
 
+// CreateVolume ...
+func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
+	logger := klog.FromContext(ctx)
+	logger.V(2).Info("starting")
+
+	// Handle static volume creation (existing fileset)
+	csiVolume, err := staticFileSetCreate(req, cs.filsetManager)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "create static volume failed: %v", err)
+	}
+	if csiVolume != nil {
+		klog.Infof("CreateVolume: static volume create successful, pvName: %s, VolumeId: %s, volumeContext: %v", req.Name, csiVolume.VolumeId, csiVolume.VolumeContext)
+		return &csi.CreateVolumeResponse{Volume: csiVolume}, nil
+	}
+
+	// Validate parameters
+	if err := validateFileSetParameters(req); err != nil {
+		klog.Errorf("CreateVolume: error parameters from input: %v, with error: %v", req.Name, err)
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid parameters from input: %v, with error: %v", req.Name, err)
+	}
+
+	// Extract parameters
+	params := req.GetParameters()
+	bmcpfsID := params["bmcpfsId"]
+	fullPath := req.Name
+	if rootPath, ok := params["path"]; ok && rootPath != "" {
+		fullPath = filepath.Join(rootPath, req.Name)
```

Comment:
Is it possible to update the path field with the final fullpath?  BMCPFS VolumePopulator needs to extract the target Path from the PV.